action/protocol/staking/handlers.go

```diff
@@ -357,6 +361,11 @@ func (p *Protocol) handleChangeCandidate(ctx context.Context, act *action.Change
 			failureStatus: iotextypes.ReceiptStatus_ErrNotEnoughBalance,
 		}
 	}
+	// clear previous candidate's self stake if the bucket is expired
```

Comment:
1. why need this change? change candidate has nothing to do with previous candidate's self-stake bucket

Reply:
we should clear self-stake bucket if the bucket is an expired endorse bucket

Reply:
then need to check the bucket is an expired endorse bucket? not a regular bucket